### ISSUE TITLE
docs for extend_from_slice say extend isnt specialized for slice yet

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust#53185]
Noticed an error in the docs of Vec::extend_from_slice:

> Note that this function is same as extend except that it is specialized to work with slices instead. If and when Rust gets specialization this function will likely be deprecated (but still available).

extend is already specialized for slices

**Comment by frewsxcv** [comment]
@rust-lang/libs Are there plans to deprecate `extend_from_slice` in favor of `extend`, or can we just remove this comment altogether?

**Comment by SimonSapin** [comment]
If I remember correctly, the plan is to start "publicly" taking advantage of specialization in the standard library (even if indirectly, such as by recommending `extend` over `extend_from_slice`) after specialization is stable in the language, because of uncertainty around what eventually will or will not be allowed by specialization.

**Comment by alexcrichton** [comment]
Yes until specialization is stable we do not want to deprecate this method.

**Comment by steveklabnik** [comment]
Seems like this is blocked on specialization, so tagging with low priority.

**Comment by briansmith** [comment]
On the other hand, the documentation for `Extend<&'a T>` says that it *is* specialized for slice iterators, and it is pretty important that it is; otherwise we'd need `extend_from_slice` to be generalized into a trait.

**Comment by Iselk** [comment]
Any news on this? Should I make a pull request depreciating the method or teak the docs?

**Comment by thomcc** [comment]
Seems like it's still blocked on specialization stabilizing...

**Comment by Iselk** [comment]
But I can’t help but notice the function (in the alloc source) turns the slice to an iterator an passes it to extend()
extend_from_slice is then an middlehand

**Comment by thomcc** [comment]
Yes, that's already been discussed.

**Comment by Iselk** [comment]
So the function exists only if specialisation would disappear for slice iterators in the future?
Are there any updates on specialisation in the last two years?

**Comment by thomcc** [comment]
Yes: https://github.com/rust-lang/rust/issues/31844

It's also possible it will go the same route as const generics with a `min_specialization` (which already exists) stabilizing first. But it might not too, and might need to wait for Chalk to be integrated into the compiler more fully.

**Comment by Iselk** [comment]
Okay.
The function docs suggest it’s not specialised, as the issue author points out, but seeing as it turns it into an iterator, the docs are miss-leading. Should we change them?

**Comment by hkBst** [comment]
Current text is:
> Note that this function is the same as [extend](https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.extend), except that it also works with slice elements that are Clone but not Copy. If Rust gets specialization this function may be deprecated.

which explains the difference and does not suggest that extend is not specialized for slices yet.

@bjorn3 can this be closed?
